allennlp/commands/train.py

```diff
@@ -172,7 +186,41 @@ def train_model(params: Params, serialization_dir: str, file_friendly_logging: b
     """
     prepare_environment(params)
 
-    os.makedirs(serialization_dir, exist_ok=True)
+    if os.path.exists(serialization_dir):
+        logger.info(f"Continuing from prior training at {serialization_dir}.")
```

Comment:
Can you pull this whole block out into its own method?  Something like `check_params_match(serialization_dir, params)`?  It makes this method not so heavy, and it is easier to test.

Reply:
Good call.  Done.